API/User/business.py

```diff
@@ -1,175 +1,210 @@
+        return ErrorCode.STREET_NOK
+    if 'town' in infos and not re.search(regex_address, infos['town']):
+        return ErrorCode.CITY_NOK
+    if 'region' in infos and not re.search(regex_name, infos['region']):
+        return ErrorCode.REGION_NOK
+    return None
+
+
+def get_user(request):
+    if not request:
+        abort(400)
+
+    if 'mail' in request.args:
+        user = session.query(User).filter(User.mail == request.args['mail']).first()
+        if not user:
+            return HttpResponse(403).error(ErrorCode.USER_NFIND)
+        res = {
+            'user': to_dict(user)
+        }
+        return HttpResponse().custom(res)
+
+    users = session.query(User).filter()
+    res = {
+        'users': [to_dict(x) for x in users]
+    }
+    return HttpResponse().custom(res)
+
+
+def delete_user(request, user_id):
+    """
+    Delete an user
+    :param request:
+    :param user_id:
+    :return: Httreponse
+    """
+    if not request:
+        abort(400)
+
+    if not check_user_permission(user_id):
+        error = {
+            'error': 'Action interdite: Tentative d\'action sur un compte non identifié'
+        }
+        return HttpResponse(403).custom(error)
+
+    user = session.query(User).filter(User.id == user_id).first()
+    if not user:
+        return HttpResponse(403).error(ErrorCode.USER_NFIND)
+    try:
+        session.query(User).filter(User.id == user_id).delete()
+        session.commit()
+    except Exception as e:
+        session.rollback()
+        session.flush()
+        return HttpResponse(500).error(ErrorCode.DB_ERROR, e)
+    return HttpResponse(202).success(SuccessCode.USER_DELETED)
+
+
+def create_user(request):
+    if not request:
+        abort(400)
+
+    verification = check_user_infos(request.json)
+    if verification is not None:
+        return HttpResponse(403).error(verification)
+
+    existing = session.query(User).filter(User.mail == request.json['mail']).first()
+
+    if existing:
+        return HttpResponse(403).error(ErrorCode.MAIL_USED)
+
+    hashed = get_hashed_password(request.json['password'])
+    new_user = User(
+        date_insert=datetime.datetime.now(),
+        date_update=datetime.datetime.now(),
+        is_active=1,
+        status=0,
+        gender=request.json['gender'],
+        lastname=request.json['lastname'],
+        firstname=request.json['firstname'],
+        mail=request.json['mail'],
+        password=hashed,
+        country=request.json['country'],
+        town=request.json['town'],
+        street=request.json['street'],
+        street_number=request.json['street_number'],
+        region=request.json['region'],
+        postal_code=request.json['postal_code']
+    )
+
+    try:
+        session.add(new_user)
+        session.commit()
+    except Exception as e:
+        session.rollback()
+        session.flush()
+        return HttpResponse(500).error(ErrorCode.DB_ERROR, e)
+
+    # token generation for new user
+    try:
```

Comment:
normal les deux try d'affilé ?

Reply:
Non, c'est corrigé sur develop mais augustin arrete pas de le récupérer de je sais pas ou >.<
https://github.com/arlex-it/arlex/blob/de6a7685f5a1e82cf9f86856692395b58f167e7d/API/User/business.py#L98

C'est censé etre comme ca. Il manque plein de trucs sur cette branche, comme les session.begin() etc

Reply:
C'est une vieille pr mais je vais corriger